src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java

```diff
@@ -169,8 +169,16 @@ private static boolean containsNullCaseLabelElement(DetailAST detailAst) {
      * @return true if part of a switch expression
      */
     private static boolean isSwitchExpression(DetailAST ast) {
-        return ast.getParent().getType() == TokenTypes.EXPR
-                || ast.getParent().getParent().getType() == TokenTypes.EXPR;
+        final int[] switchStatementParents = {
```

Comment:
> Also move this to a field constant.

Done. has to make it a fileStateful to do this

Reply:
It shouldn't. This field isn't mutated, so its immutable.

What test is requiring you to make the change?

Reply:
`testClassesWithMutableFieldsShouldBeStateful` and  `testFieldsInStatelessChecksShouldBeImmutable` in `ImmutabilityTest`.

```
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes that do not have immutable fields and do not have modifier ABSTRACT should be annotated with @FileStatefulCheck or should be annotated with @GlobalStatefulCheck or should be suppressed' was violated (1 times):
Class <com.puppycrawl.tools.checkstyle.checks.coding.MissingSwitchDefaultCheck> is not annotated with @FileStatefulCheck in (MissingSwitchDefaultCheck.java:0) and Class <com.puppycrawl.tools.checkstyle.checks.coding.MissingSwitchDefaultCheck> is not annotated with @GlobalStatefulCheck in (MissingSwitchDefaultCheck.java:0) and should be suppressed or resolved.
```


```
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'fields that are declared in classes that are annotated with @StatelessCheck and are not module properties should be among immutable types and should be final or should be suppressed' was violated (1 times):
Field <com.puppycrawl.tools.checkstyle.checks.coding.MissingSwitchDefaultCheck.SWITCH_STATEMENT_PARENTS> should be among immutable types in (MissingSwitchDefaultCheck.java:0) and should be suppressed or resolved.
```

 I don't know why we see int arrays as mutable.

Reply:
Technically they are. You can do `SWITCH_STATEMENT_PARENTS[0] = -1;` and it will work.

This is an akward issue. I would say to use a list, but that would require us to create a new method since it takes only arrays. I am surprised we haven't run into this issue before.

I am ok to restore the final local variable, but please create an issue on this.

Reply:
done. restored to a local variable

Reply:
https://github.com/checkstyle/checkstyle/issues/15541